### ISSUE TITLE
pylint fails to resolve typing.Self from base classes

### ISSUE THREAD

**Issue by kdkavanagh** [pylint-dev/pylint#10807]
Very similar to #9159, pylint is unable to resolve `typing.Self` from base classes even when the method in question is not overridden (as it is in #9159)

### Bug description

```python


import typing


class A:
    def parent_meth(self) -> typing.Self:
        return self


class B(A):
    def meth(self) -> typing.Self:
        return super().parent_meth()

    def f2(self):
        pass


B().meth().f2()
```

### Configuration

```ini

```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:4:0: R0903: Too few public methods (1/2) (too-few-public-methods)
test.py:17:0: E1101: Instance of 'A' has no 'f2' member (no-member)
```

### Expected behavior

`B().meth()` should resolve to type `B`, thereby exposing `f2()`

### Pylint version

```shell
$ pylint --version
pylint 4.0.2
astroid 4.0.2
Python 3.12.6 (main, Oct  9 2024, 11:19:43) [GCC 13.2.0]
```

### Additional dependencies

```python

```

**Comment by kdkavanagh** [comment]
Confirmed this is an issue even without the `typing.self` annotations

**Comment by kdkavanagh** [comment]
Also related: https://github.com/pylint-dev/astroid/issues/2852

**Comment by kdkavanagh** [comment]
I believe this was fixed in astroid v4.1.0, https://github.com/pylint-dev/astroid/pull/2936
